app/controllers/home_controller.rb

```diff
@@ -20,13 +21,15 @@ def create
   def show
     @link = Link.find(params[:id])
     @comment = Comment.new
-    @comments = Comment.all
+    @comments = @link.comments.all
     flash[:back] = home_path(params[:id])
   end
 
   def comment_create
+    @link = Link.find(params[:comment][:commentable])
     @comment = Comment.new(params_comment)
```

Comment:
we can go with `current_user.comment.new(params_comment)`

Reply:
sure, i will do that